src/builder.ts

```
@@ -95,6 +95,11 @@ export function clientSupportsStdio(clientId: ClientId): boolean {
  * @returns The CLI command string, or null if the client doesn't support CLI installation
  */
 export function buildCommand(clientId: ClientId, serverData: GleanServerConfig): string | null {
-  const builder = registry.createBuilder(clientId);
-  return builder.buildCommand(serverData);
+  try {
+    const builder = registry.createBuilder(clientId);
+    return builder.buildCommand(serverData);
+  } catch (error) {
+    // Return null for any errors (unsupported clients, validation errors, etc.)
+    return null;
```

1/5 (nit, non-blocking)

For a future iteration, we might want to emit a warning or log message or something (silently doing nothing might end up trolling us in odd ways).

---

I was considering this too. Happy to update this too once we figure out how we want to handle this.